be/test/vec/data_types/data_type_array_test.cpp

```diff
@@ -0,0 +1,490 @@
+        Field default_field_array = Array();
+        create_column_assert(type, default_field_array, 24);
+    }
+    // for decimal32/64/128/256 here uncompressed size is 16
+    for (int i = 14; i < array_types.size(); i++) {
+        auto type = remove_nullable(array_types[i]);
+        Field default_field_array = Array();
+        create_column_assert(type, default_field_array, 16);
+    }
+}
+
+TEST_F(DataTypeArrayTest, GetFieldTest) {
+    TExprNode node;
+    node.node_type = TExprNodeType::ARRAY_LITERAL;
+    for (size_t i = 0; i < array_columns.size(); ++i) {
+        Field assert_field;
+        array_columns[i]->get(0, assert_field);
+        get_field_assert(array_types[i], node, assert_field, true);
+    }
+}
+
+TEST_F(DataTypeArrayTest, FromAndToStringTest) {
+    // insert from data csv and assert insert result
+    for (int i = 0; i < array_columns.size(); i++) {
+        auto& column = array_columns[i];
+        auto& type = array_types[i];
+        std::cout << "type: " << type->get_name() << " for column size: " << column->size()
+                  << std::endl;
+        // datatype array<string> has some different behavior maybe wrong with given data
+        if (i == 13) {
+            continue;
+        }
+        assert_to_string_from_string_assert(column->assume_mutable(), type);
+    }
+}
+
+TEST_F(DataTypeArrayTest, CompareTest) {
+    for (int i = 0; i < array_types.size(); i++) {
+        auto& type = array_types[i];
+        auto array_type = assert_cast<const DataTypeArray*>(remove_nullable(type).get());
+        auto nested_type = array_type->get_nested_type();
+        EXPECT_EQ(nested_type->is_comparable(), array_type->is_comparable());
+        EXPECT_FALSE(array_type->equals(*type));
+    }
+}
+
+TEST_F(DataTypeArrayTest, SerdeHiveTextAndJsonFormatTest) {
+    // insert from data csv and assert insert result
+    for (int i = 0; i < array_types.size(); i++) {
+        MutableColumns array_cols;
+        array_cols.push_back(array_columns[i]->get_ptr());
+        CommonDataTypeSerdeTest::load_data_and_assert_from_csv<true, true>({serdes[i]}, array_cols,
+                                                                           data_files[i], ';');
+        CommonDataTypeSerdeTest::load_data_and_assert_from_csv<false, true>({serdes[i]}, array_cols,
+                                                                            data_files[i], ';');
+    }
+}
+
+TEST_F(DataTypeArrayTest, SerdePbTest) {
+    // fix serde pb for read decimal64 not support
+    CommonDataTypeSerdeTest::assert_pb_format(array_columns, serdes);
+}
+
+TEST_F(DataTypeArrayTest, SerdeJsonbTest) {
+    CommonDataTypeSerdeTest::assert_jsonb_format(array_columns, serdes);
+}
+
+TEST_F(DataTypeArrayTest, SerdeMysqlTest) {
+    // insert from data csv and assert insert result
+    CommonDataTypeSerdeTest::assert_mysql_format(array_columns, serdes);
+}
+
+TEST_F(DataTypeArrayTest, SerdeArrowTest) {
+    // todo. fix decimal256 serde
```

Comment:
What's the problem? It should be fixed before this test.

Reply:
not implement